src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodFilteringService.cs

```diff
@@ -0,0 +1,191 @@
+                new object[] { project.Id, targetTypeNames.ToArray(), loadOnly },
+                cancellationToken).ConfigureAwait(false);
+
+            if (!remoteResults.Any())
+            {
+                return null;
+            }
+
+            // Reconstruct the dictionary from remote results.
+            var results = new MultiDictionary<string, string>();
+
+            foreach (var pair in remoteResults)
+            {
+                foreach (var value in pair.Item2)
+                {
+                    results.Add(pair.Item1, value);
+                }
+            }
+
+            return results;
+        }
+
+        private static async Task<MultiDictionary<string, string>?> GetPossibleExtensionMethodMatchesInCurrentProcessAsync(Project currentProject, ImmutableHashSet<string> targetTypeNames, bool loadOnly, CancellationToken cancellationToken)
+        {
+            var solution = currentProject.Solution;
+            var graph = currentProject.Solution.GetProjectDependencyGraph();
+            var relevantProjectIds = graph.GetProjectsThatThisProjectTransitivelyDependsOn(currentProject.Id)
+                                        .Concat(currentProject.Id);
+
+            using var syntaxDisposer = ArrayBuilder<SyntaxTreeIndex>.GetInstance(out var syntaxTreeIndexBuilder);
+            using var symbolDisposer = ArrayBuilder<SymbolTreeInfo>.GetInstance(out var symbolTreeInfoBuilder);
+            var peReferences = PooledHashSet<PortableExecutableReference>.GetInstance();
+
+            try
+            {
+                foreach (var projectId in relevantProjectIds.Concat(currentProject.Id))
+                {
+                    // Alway create indices for documents in current project if they don't exist.
+                    loadOnly &= projectId != currentProject.Id;
+
+                    var project = solution.GetProject(projectId);
+                    if (project == null || !project.SupportsCompilation)
+                    {
+                        continue;
+                    }
+
+                    // Transitively get all the PE references
+                    peReferences.AddRange(project.MetadataReferences.OfType<PortableExecutableReference>());
+                    foreach (var document in project.Documents)
+                    {
+                        // Don't look for extension methods in generated code.
+                        if (document.State.Attributes.IsGenerated)
+                        {
+                            continue;
+                        }
+
+                        var info = await document.GetSyntaxTreeIndexAsync(loadOnly, cancellationToken).ConfigureAwait(false);
+
+                        // Don't provide anyting if we don't have all the required SyntaxTreeIndex created.
+                        if (info == null)
+                        {
+                            return null;
+                        }
+
+                        if (info.ContainsExtensionMethod)
```

Comment:
this is interesting.  so you're examining the syntax tree info to determine if the symbol tree info should be examined...

Why not just have the symboltreeinfo contain this bit?

Reply:
Ah.  my guess is: this is a quick syntactic way to proceed.  In which case: ok.

Reply:
No, this is to determine if syntax tree index should be examined, and there's another check for symbol tree info a few lines below. (and yes, symboltreeinfo also has this flag)